cf-custom-resources/lib/dns-cert-validator.js

```diff
@@ -293,19 +281,28 @@ const deleteHostedZoneRecords = async function (
       // we'll remove validation CNAME records only for app and root hosted zone,
       // since the legacy cert still needs the validation record in the env hosted zone.
       for (const option of options) {
-        if (option.DomainName === `${envName}.${appName}.${domainName}`) {
+        if (option.DomainName === certDomain || option.DomainName === `*.${certDomain}`) {
```

Comment:
OK so 
1. `option.DomainName` is not necessarily  `certificate.DomainName` right? 
   1. it can be a subject alternative name?
2.  This check over here ensures that we don't delete the validation record for the env hosted zone so that the other certificates in the environment remain valid. Is that correct?

Reply:
1. no they are different thing. yes it can be a SAN domain.
2. yes exactly.